tools/confix/cmd/diff.go

```diff
@@ -3,44 +3,45 @@ package cmd
 import (
 	"errors"
 	"fmt"
+	"strings"
 
+	"github.com/cosmos/cosmos-sdk/client"
 	"github.com/spf13/cobra"
 	"golang.org/x/exp/maps"
 
 	"cosmossdk.io/tools/confix"
-
-	"github.com/cosmos/cosmos-sdk/client"
 )
 
+// DiffCommand creates a new command for comparing configuration files
 func DiffCommand() *cobra.Command {
 	return &cobra.Command{
-		Use:   "diff [target-version] <app-toml-path>",
-		Short: "Outputs all config values that are different from the app.toml defaults.",
-		Args:  cobra.MinimumNArgs(1),
+		Use:   "diff [target-version] <config-path> [config-type]",
+		Short: "Outputs all config values that are different from the default.",
+		Long:  "This command compares the specified configuration file (app.toml or client.toml) with the defaults and outputs any differences.",
+		Args:  cobra.MinimumNArgs(2),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			var filename string
+			targetVersion := args[0]
+			configPath := args[1]
+			configType := confix.AppConfigType // Default to app configuration
 			clientCtx := client.GetClientContextFromCmd(cmd)
-			switch {
-
-			case len(args) > 1:
-				filename = args[1]
-			case clientCtx.HomeDir != "":
-				filename = fmt.Sprintf("%s/config/app.toml", clientCtx.HomeDir)
-			default:
-				return errors.New("must provide a path to the app.toml file")
+
+			if len(args) > 2 {
+				configType = strings.ToLower(args[2])
+				if configType != confix.AppConfigType && configType != confix.ClientConfigType {
```

Comment:
Please hoist this validation out of this if statement and put it down after it so that there will be unconditional validation of the configType.